pkg/pxe/schemes.go

```diff
@@ -216,34 +246,27 @@ func (h HTTPClientWithRetries) GetFile(u *url.URL) (io.ReaderAt, error) {
 		return nil, err
 	}
 
-	counter := 0
-	delay := 500 * time.Millisecond
-	maxDelay := 10 * time.Second
-
-	timer := time.NewTimer(h.Timeout)
-	defer timer.Stop()
-	for {
-		resp, err := h.Client.Do(req)
-		if err == nil {
-			if resp.StatusCode != 200 {
-				return nil, fmt.Errorf("HTTP server responded with code %d, want 200: response %v",
-					resp.StatusCode, resp)
-			}
-			return uio.NewCachingReader(resp.Body), nil
+	h.BackOff.Reset()
+	for d := time.Duration(0); d != backoff.Stop; d = h.BackOff.NextBackOff() {
+		if d > 0 {
+			time.Sleep(d)
 		}
-		counter++
 
-		// Backoff
-		delay += delay / 2
-		if delay > maxDelay {
-			delay = maxDelay
+		var resp *http.Response
+		// Note: err uses the scope outside the for loop.
+		resp, err = h.Client.Do(req)
```

Comment:
In the above version you use 

```go
if err == nil {
  return resp.Body...
}
Error handling
```

And this one is the opposite. Mind making them the same pattern?